apps/web/src/hooks/useRegisterNameCallback.ts

```diff
@@ -22,10 +27,46 @@ export function useRegisterNameCallback(
   value: bigint | undefined,
   years: number,
   discountKey?: `0x${string}`,
-  validationData?: `0x${string}`,
-) {
-  const { address } = useAccount();
+  validationData?: `0x${string}`,) {
+  if (discountKey) {
+    console.log('discountKey', discountKey);
+    console.log('validationData', validationData);
+  }
+  const { address, chainId } = useAccount();
+  
+  const account = useAccount();
+  const { data: client } = useWalletClient();
+  const { writeContract } = useWriteContract();
+  const network = chainId === baseSepolia.id ? chainId : base.id;
+  const { writeContractsAsync } = useWriteContracts(); // For Smart Contract Wallet calls
+  
+  
+  const {data: availableCapacities} = useCapabilities({
+      account: address
+    });
+  
+  const capabilities = useMemo(() => {
+    if (!account.isConnected || !chainId || !availableCapacities) {
+      return {};
+    }
+    const chainCapabilities = availableCapacities[chainId];
+    if (chainCapabilities["paymasterService"] && chainCapabilities["paymasterService"].supported) {
+      return {
+        paymasterService: {
+          // url: `${document.location.origin}/api/paymaster`
```

Comment:
what's the significance of this comment?

Reply:
Might have to uncomment that before deployment.

Reply:
if we need to do anything before production to ensure this works, make sure it's documented sufficiently (i.e., punchlist item, linear, etc. wherever it's relevant)